ui/src/components/Map/MapSearch.vue

```diff
@@ -63,6 +65,33 @@ const results = computed(() => {
   }
   return []
 })
+
+// search term set by an outside component rather than user text input, i.e. from MapNodesGrid
+const nodeSearchTerm = computed<string>(() => store.state.mapModule.nodeSearchTerm)
+
+watch(nodeSearchTerm, async (searchTerm) => {
```

Comment:
Changed the `watch(nodeSearchTerm)` to use `watchEffect` and added comments to explain what it is doing.

In case of other `watch(results)`, needed to keep as `watch` since it should only be triggered by `results` and not by `outsideSearch`. Factored out into separate method + comments to make it more intelligible.

Basically what this is doing is allowing when a user clicks on the "node table" below the map on a node ID or node Label, this will auto-populate the `MapSearch` autocomplete as if the user had typed it in; and once the search is done it selects the first result and "flies" to the node marker on the map.